### PR TITLE
Group passed recordings of flaky tests with the failed ones

### PR DESCRIPTION
**Pull request by hbenl**
but show them with a "Passed" label so that they can be distinguished from the failing ones for the same test.

### CODE REVIEW

**Comment by bvaughn** [comment]
> Group passed recordings of flaky tests with the failed ones

There should be up to three groups of tests:

1. Failed
2. Flaky
3. Passing

Each test should only appear in one of the groups. (If a test had some passing runs and some failing runs, then ti should be in the flaky group– not the failed group.)

**Review by bvaughn — COMMENTED** [review]
Can we sort test attempts in the tree by time so that the passing tests always come after/below the flaky ones? It's confusing at the moment I think.

Screen Shot 2023-09-29 at 9 54 51 AM

**Review by bvaughn — APPROVED** [review]
I think this is an improvement

tag @jonbell-lot23 for his thoughts though

File: src/ui/utils/testRuns.ts [hunk 1]
```diff
@@ -77,34 +77,45 @@ export function groupRecordings(recordings: Recording[]): RecordingGroups {
...
-            failedRecordings.count++;
-            failedRecordings.fileNameToRecordings[filePath] = [recording];
-          }
+          addToRecordingGroup(failedRecordings, recording);
         }
       }
     }
+
+    if (didAnyTestFail && didAnyTestPass) {
+      flakyRecordings.fileNameToRecordings[filePath].sort((recordingA, recordingB) => {
```
Comment (bvaughn) [inline src/ui/utils/testRuns.ts:109]: Why isn't sorting by recording date/time sufficient? (Do we not have that info loaded?)

Comment (hbenl) [reply src/ui/utils/testRuns.ts:109]: IIRC the recording date/time is the time when the upload started and (again, IIRC) that's not reliably in the same order.

Comment (bvaughn) [reply src/ui/utils/testRuns.ts:109]: Ooof. That's fair.

**Pull request merged**